### ISSUE TITLE
feat(deletion): delete orphan Opinions, Notes and Tasks

### ISSUE THREAD

**Issue by SamuelHassine** [LuatixHQ/opencti#3777]
## Use case

Add a built-in Retention policy to remove Notes, Opinions and Tasks that have been left orphans after entities deletion.

**Comment by Jipegien** [comment]
Cascade delete pop up used in report deletion ![image](https://github.com/OpenCTI-Platform/opencti/assets/104078945/8cc545a9-f217-49a6-879b-a9aed9c3c354)

**Comment by nino-filigran** [comment]
@SamuelHassine & @Jipegien 
My gut feeling is that this should be a built-in policy, that the user can modify (enabled/disabled and to modify the retention period). Would you agree with this approach? Or would you expect it as an automation, that the user can enable/disable? 

If we go with the first approach, in essence, we would need to add a filter to detect if an element is an orphan or not, and the possibility enable/disable a policy. Would that cover your need?

**Comment by SamuelHassine** [comment]
Yes @nino-filigran, please reach out @richard-julien we had a discussion about this.

**Comment by SamuelHassine** [comment]
Built-in policies which can be enabled / disabled by the administrator.

**Comment by nino-filigran** [comment]
Note from a meeting with @Jipegien : notes + tasks contain entities. Therefore, we should also check that they are empty.
